src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs

```diff
@@ -216,20 +216,20 @@ public string FindAssemblyFile (AssemblyNameReference reference)
 				name + ".dll");
 		}
 
-		public AssemblyDefinition Resolve (AssemblyNameReference reference, ReaderParameters parameters)
+		public AssemblyDefinition Resolve (AssemblyNameReference reference, ReaderParameters? parameters)
 		{
 			var name = reference.Name;
 
 			AssemblyDefinition assembly;
 			if (cache.TryGetValue (name, out assembly))
 				return assembly;
 
-			string assemblyFile;
-			AssemblyDefinition candidate = null;
+			string? assemblyFile;
+			AssemblyDefinition? candidate = null;
 			foreach (var dir in SearchDirectories) {
 				if ((assemblyFile = SearchDirectory (name, dir)) != null) {
 					var loaded = Load (assemblyFile);
-					if (Array.Equals (loaded.Name.MetadataToken, reference.MetadataToken))
+					if (Array.Equals (loaded!.Name.MetadataToken, reference.MetadataToken))
```

Comment:
I don't understand this change: `Load()` was updated to return `AssemblyDefinition?`, implying that `null` *can* be returned, so why do we "assert" that `null` *won't* be returned here?  Because `assemblyFile` isn't null?